src/main/kotlin/com/sourcegraph/cody/agent/protocol/ChatMessage.kt

```diff
@@ -1,10 +1,44 @@
 package com.sourcegraph.cody.agent.protocol
 
+import java.time.OffsetDateTime
+
+data class ChatError(
+    val kind: String? = null,
+    val name: String,
+    val message: String,
+    val retryAfter: String? = null,
+    val limit: Int? = null,
+    val userMessage: String? = null,
+    val retryAfterDate: OffsetDateTime? = null,
+    val retryMessage: String? = null,
+    val feature: String? = null,
+    val upgradeIsAvailable: Boolean? = null,
+) {
+  fun toRateLimitError(): RateLimitError? {
+    if (this.retryAfter == null ||
+        this.limit == null ||
+        this.userMessage == null ||
+        this.retryMessage == null ||
+        this.feature == null ||
+        this.upgradeIsAvailable == null) {
+      return null
+    }
+    return RateLimitError(
```

Comment:
Do we not copy all the parameters, e.g. `feature`?

Reply:
afaik: `feature` is either "autocomplete" or "chat" (or smth else?) 

we do not need them. I simplified the API in this scope